convert2rhel/actions/pre_ponr_changes/backup_system.py

```diff
@@ -87,9 +87,10 @@ def backup_yum_repos(self):
         loggerinst.info("Backing up .repo files from %s." % DEFAULT_YUM_REPOFILE_DIR)
 
         repo_files_backed_up = False
-
         for repo in os.listdir(DEFAULT_YUM_REPOFILE_DIR):
-            if repo.endswith(".repo") and repo != "redhat.repo":
+            if (repo.endswith(".repo") and repo != "redhat.repo") or (
+                subscription.should_subscribe and repo == "redhat.repo"
+            ):
```

Comment:
What about rewriting the loop like this? 

```python
        for repo in os.listdir(DEFAULT_YUM_REPOFILE_DIR):
            # backing up redhat.repo so repo files are properly backed up when doing satellite conversions

            if not repo.endswith(".repo"):
                # Maybe log that the file is not repo?
                return

            if not subscription.should_subscribe and repo == "redhat.repo":
                return

            repo_path = os.path.join(DEFAULT_YUM_REPOFILE_DIR, repo)
            restorable_file = RestorableFile(repo_path)
            backup.backup_control.push(restorable_file)
            repo_files_backed_up = True

```

Trying to simplify this check down to two things: 
1. If it does not end with `.repo`, we can bail out as it doesn't interest to us
2. If `should_subscribe` is false, and repo is `redhat.repo`, then we return
3. Otherwise, we back it up. By default, I believe `should_subscribe` is `False`, unless we detect the system is registered before the analysis/conversion. 


Does it make sense to you, @pr-watson and @SpyTec?

Reply:
Also, if possible, can we move the contents of this function (`backup_yum_repos`) to be inside the `run` method? I don't think this function is long enough to need an function like this, especially because `run` only calls this and exit. 

To me, it would be more clean if `run` performs the operations instead of calling the intermediate function.

Reply:
It looks much cleaner now - I agree with this change. And I don't mind moving backup_yum_repos into the run method

Reply:
Looks a lot better, thanks @r0x0d for the suggestion!